pkg/local_object_storage/metabase/metadata.go

```diff
@@ -0,0 +1,796 @@
+				}
+				if dbValInt == nil {
+					if len(dbVal) != intValLen {
+						dbValInt = new([]byte)
+					} else {
+						// do the same as for primary attribute, but unlike there, here we don't know
+						// whether the attribute is expected to be integer or not.
+						dbValInt = new([]byte)
+						if attrSkr.isInt(id, attr, dbVal) {
+							var err error
+							if *dbValInt, err = restoreIntAttributeVal(dbVal); err != nil {
+								return nil, nil, invalidMetaBucketKeyErr(primKey, fmt.Errorf("invalid integer value: %w", err))
+							}
+						}
+					}
+				}
+				var checkedDBVal []byte
+				if isNumericOp(m) {
+					if *dbValInt == nil {
+						continue nextPrimKey
+					}
+					checkedDBVal = dbVal
+				} else if *dbValInt != nil {
+					checkedDBVal = *dbValInt
+				} else {
+					checkedDBVal = dbVal
+				}
+				checkedDBVal, fltVal, err := combineValues(attr, checkedDBVal, fs[j].Value()) // TODO: deduplicate DB value preparation
+				if err != nil {
+					return nil, nil, invalidMetaBucketKeyErr(primKey, fmt.Errorf("invalid attribute %s value: %w", attr, err))
+				}
+				if !matchValues(checkedDBVal, m, fltVal) {
+					continue nextPrimKey
+				}
+			}
+		}
+		// object matches, collect attributes
+		collected := make([]string, len(attrs))
+		var primDBVal []byte
+		var insertI uint16
+		if len(attrs) > 0 {
+			var err error
+			if primDBVal, err = attrSkr.get(id, attrs[0]); err != nil {
+				return nil, nil, err
+			}
+			if cursor != nil { // can be < than previous response chunk
+				if c := bytes.Compare(primDBVal, prevResPrimVal); c < 0 || c == 0 && bytes.Compare(id, prevResOID) <= 0 {
+					continue nextPrimKey
+				}
+				// note that if both values are integers, they are already sorted. Otherwise, the order is undefined.
+				// We could treat non-int values as < then the int ones, but the code would have grown huge
+			}
+			for i := range n {
+				if c := bytes.Compare(primDBVal, collectedPrimVals[i]); c < 0 || c == 0 && bytes.Compare(id, res[i].ID[:]) < 0 {
+					break
+				}
+				if insertI++; insertI == count {
+					more = true
+					continue nextPrimKey
+				}
+			}
+			if collected[0], err = attrSkr.restoreVal(id, attrs[0], primDBVal); err != nil {
+				return nil, nil, err
+			}
+		} else {
+			if cursor != nil { // can be < than previous response chunk
+				if bytes.Compare(id, prevResOID) <= 0 {
+					continue nextPrimKey
+				}
+			}
+			for i := insertI; i < n; i++ {
+				if bytes.Compare(id, res[i].ID[:]) >= 0 {
+					if insertI++; insertI == count {
+						more = true
+						continue nextPrimKey
+					}
+				}
+			}
+		}
+		for i := 1; i < len(attrs); i++ {
+			val, err := attrSkr.get(id, attrs[i])
+			if err != nil {
+				return nil, nil, err
+			}
+			if collected[i], err = attrSkr.restoreVal(id, attrs[i], val); err != nil {
+				return nil, nil, err
+			}
+		}
+		if n == count {
+			more = true
+			if len(attrs) > 0 {
+				break
+			} // else "later" objects may have "less" ID, so we should continue
```

Comment:
But how? OIDs are sorted naturally as well and if have enough results then subsequent IDs should be probed on subsequent (continuation) search queries.

Reply:
consider objects with ascending index and descending IDs. If no attributes are requested, we should sort by IDs. We can state that first _filtered_ attribute takes order priority in this case. If so, we can always break here

Reply:
OK, let's leave it as is for now.